client/state/sites/selectors/has-site-product-jetpack-stats.ts

```diff
@@ -11,13 +13,22 @@ const hasSiteProductJetpackStats = (
 	onlyPaid = false,
 	siteId = getSelectedSiteId( state )
 ): boolean => {
+	const isWpcomSite = !! isSiteWpcom( state, siteId );
+
 	const siteHasStatsProduct = getSitePurchases( state, siteId )?.some(
 		( product ) =>
 			productHasStats( camelOrSnakeSlug( product ), onlyPaid ) && product.expiryStatus !== 'expired'
 	);
+
 	const sitePlan = getSitePlan( state, siteId );
 	const siteHasStatsPlan = productHasStats( sitePlan?.product_slug as string, onlyPaid );
-	return siteHasStatsPlan || !! siteHasStatsProduct;
+
+	// Check whether sites have paid plans of WPCOM.
+	const siteHasPaidPlan = isSiteOnPaidPlan( state, siteId || 0 );
```

Comment:
As we talked on Slack, the logic here is some interim solution, so I'm thinking, could we move these out to the notices and not to change anything here? The reason is the file could be just the same even after we bundled Stats into WPCOM plans.

Reply:
Moved in https://github.com/Automattic/wp-calypso/pull/81514/commits/2e5035da810bdc99362937f3af6b287ec1714941.